test.c

```diff
@@ -0,0 +1,72 @@
+#include "tree.h"
```

Comment:
It's not possible to valgrind Go binaries, so I wrote this program and caught a few memory leaks in `tree.c` with it. We are now leakless.